pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/buffer/TransactionBuffer.java

```diff
@@ -188,14 +188,14 @@ public interface TransactionBuffer {
     TransactionBufferStats getStats(boolean lowWaterMarks);
 
     /**
-     * Wait TransactionBuffer Recovers completely.
-     * Take snapshot after TB Recovers completely.
+     * Wait TransactionBuffer recovers completely.
      * @param isTxn
-     * @return a future which has completely if isTxn = false. Or a future return by takeSnapshot.
+     * @return a future whether the transaction buffer recover completely.
      */
-    CompletableFuture<Void> checkIfTBRecoverCompletely(boolean isTxn);
-
-
+    @Deprecated
+    default CompletableFuture<Void> checkIfTBRecoverCompletely(boolean isTxn) {
+        return CompletableFuture.completedFuture(null);
+    }
```

Comment:
Same as the above comment.

Reply:
Recently, we found this method will be helpful in some cases where the caller needs to wait for the transaction buffer to recover completely. For example, in https://github.com/apache/pulsar/pull/21466, the persistent topic needs to wait for the transaction buffer to recover completely and then get the max read position.

But the method signature is deprecated because the param `boolean isTxn` is useless. We remove this method and replace it with a new method  `CompletableFuture<Void> checkIfTBRecoverCompletely()`.
But I have a concern about this, we provide `TransactionBufferProvider` for the users to implement a custom version based on the `TransactionBuffer` interface. Will this change impact these users?

Reply:
> But I have a concern about this, we provide TransactionBufferProvider for the users to implement a custom version based on the TransactionBuffer interface. Will this change impact these users?

It's a beta interface, which means it is not stable. If the method doesn't make sense to the interface, then we can remove it.
 
https://github.com/apache/pulsar/pull/21406/files/5e4dc0be9f1c7e028b37b69a4be929fcac11a631#diff-819cb066b87a312a85c161d7bccf5699ddd8e1aded6545a6f39a237d16e6e707L51